sdk/src/identity/claim_aggregation/w3c_vc/did.rs

```diff
@@ -122,6 +122,12 @@ impl DidBuf {
         }
     }
 
+    // mismatched_lifetime_syntaxes reproting an error we are creating a
+    // [Did] from a [&str] reference.
+    //
+    // We are allowing "unknown_lints" because "mismatched_lifetime_syntaxes"
+    // is only in nightly.
+    #[allow(unknown_lints, mismatched_lifetime_syntaxes)]
```

Comment:
We may want to consider changing the `Did::new_unchecked` API to take a `String` instead?